scripts/make_libmuscle_api.py

```diff
@@ -1026,13 +1041,14 @@ def __copy__(self) -> 'Elements':
             "KEEPS_NO_STATE_FOR_NEXT_USE",
             "STATE_NOT_REQUIRED_FOR_NEXT_USE"])
 
+
 instance_desc = Class(
-        'Instance', None, [instance_constructor] + [
+        'Instance', None, [cast(Member, instance_constructor)] + [
```

Comment:
Why do we need a cast? It's a `Constructor`, which is a subclass of `Member`, right?

Reply:
Because `[instance_constructor]` is of type `List[Constructor]` and you're not allowed to add a `List[Member]` to it:

```
scripts/make_libmuscle_api.py:1046: error: Argument 3 to "Class" has incompatible type "List[Constructor]"; expected "List[Member]"
scripts/make_libmuscle_api.py:1046: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
scripts/make_libmuscle_api.py:1046: note: Consider using "Sequence" instead, which is covariant
scripts/make_libmuscle_api.py:1047: error: List comprehension has incompatible type List[Member]; expected List[Constructor]
```

Therefore I explicitly cast it to `Member` before putting it in the list.

Reply:
Ah of course! Makes sense now.